tests/publiccloud/create_aistack_env.pm

```diff
@@ -158,14 +177,41 @@ sub install_aistack_chart {
     }    #pod loop
 
     assert_script_run("kubectl get all --namespace $namespace");
-    record_info("Logs for the pods which is not in running or pending state");
-    foreach my $pod (@out) {
-        my $status = script_output("kubectl get pod $pod -n $namespace -o=jsonpath='{.status.phase}'", proceed_on_failure => 1);
-        if ($status !~ /^(Running|Pending|Completed)$/) {
-            my $logs = script_output("kubectl logs $pod -n $namespace", proceed_on_failure => 1);
-            record_info("$pod is in $status state. Logs:\n$logs\n");
-        }
-    }    # pod loop
+    if (@failed_pods) {
+        record_info("Failed pods:", join("\n", map { "$_->{name}: $_->{status}" } @failed_pods));
+        exit 1;
```

Comment:
Please, all `exit` calls meaning **test failed**, shall land in a `die(...)` function also needed to trigger "post-fail" cleaning routines.

Then, better to change `exit 1` to `die ...`, here and everywehere similarly needed.

Reply:
what is there for post-fail cleaning routine is expected do ? I thought the ssh_interactive_end is tearing down the terraform. That should take care of the post-fail cleaning.

Reply:
I explained in a DM the `method` inheritance from the parent classs, where the post_fail_hook is implemented: I hope it is useful :slightly_smiling_face:.